js/index.js

```diff
@@ -43,26 +43,81 @@ function renderPepperoni() {
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
-}
+ document.querySelectorAll('.mushroom').forEach(onMush =>{
+   if (state.mushrooms) {
+     onMush.style.visibility = "visible";
+    }  else {
+       onMush.style.visibility = "hidden";
+     }
+   });
+ }
+// Iteration 1: set the visibility of `<section class="mushroom">`
+
 
 function renderGreenPeppers() {
+  document.querySelectorAll('.green-pepper').forEach(onGreenPap => {
+    if (state.greenPeppers){
+      onGreenPap.style.visibility = 'visible';
+    } else {
+      onGreenPap.style.visibility = 'hidden';
+    }
+  });
   // Iteration 1: set the visibility of `<section class="green-pepper">`
 }
 
 function renderWhiteSauce() {
+  const whiteSauce = document.querySelector('.sauce');
+if(state.whiteSauce){
+  whiteSauce.classList.add('sauce-white')
+}
+else {
+  whiteSauce.classList.remove('sauce-white');
+}
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
 }
 
 function renderGlutenFreeCrust() {
+  const glutenFreeCrust = document.querySelector(".crust");
+    if(state.glutenFreeCrust){
+      glutenFreeCrust.classList.add('crust-gluten-free');
+
+    }
+else {
+  glutenFreeCrust.classList.remove('crust-gluten-free');
+}  
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 }
 
 function renderButtons() {
+const buttons = document.querySelectorAll('.panel.control');
+buttons.toggleClass('active');
+
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
 }
 
 function renderPrice() {
+  function calcPrice(){
+    
+    var totalPrice=10;
+  
+    if ($('.btn-pepperonni').hasClass('active')) {
+      totalPrice += 1;
+    }
+    if ($('.btn-mushrooms').hasClass('active')) {
+      totalPrice += 1;
+    }
+    if ($('.btn-green-peppers').hasClass('active')) {
+      totalPrice += 1;
+    }
+    if ($('.btn-sauce').hasClass('active')) {
+      totalPrice += 3;
+    }
+    if ($('.btn-crust').hasClass('active')) {
+      totalPrice += 5;
+    }
+  
+    $('#totalPrice').html(totalPrice);
+  }
   // Iteration 4: change the HTML of `<aside class="panel price">`
 }
```

Comment:
For this function you can take a look at Jan solution:


function renderPrice() {
  // Iteration 4: change the HTML of `<aside class="panel price">`
  let price = basePrice;
  let list = '';
  for (let ingredient in state) {
    if (state[ingredient]) {
      list += `<li>$ ${ingredients[ingredient].price} ${ingredients[ingredient].name}</li>`;
      price += ingredients[ingredient].price;
    }
  }
  document.querySelector('.panel.price strong').innerText = `$${price}`;
  document.querySelector('.panel.price ul').innerHTML = list;
}